core/lib/server-toast.ts

```diff
@@ -0,0 +1,84 @@
+import { cookies } from 'next/headers';
+import { ToasterProps } from 'sonner';
+import { z } from 'zod';
+import 'server-only';
+
+interface ServerToastOptions {
+  description?: string;
+  position?: ToasterProps['position'];
+}
+
+export interface ServerToastData extends ServerToastOptions {
+  message: string;
+  variant: 'success' | 'error' | 'warning' | 'info';
+}
+
+const serverToastSchema = z.object({
+  message: z.string(),
+  description: z.string().optional(),
+  variant: z.enum(['success', 'error', 'warning', 'info']),
+  position: z
+    .enum(['top-left', 'top-right', 'top-center', 'bottom-left', 'bottom-right', 'bottom-center'])
+    .optional(),
+});
+
+const TOAST_COOKIE = 'toast-notification';
+const getExpiry = () => new Date(Date.now() + 5000);
```

Comment:
Expiration needs to be instant (i.e., `maxAge: 0`) to guarantee the cookie is set only for the request right after the notification is set to avoid duplication of messages

Reply:
@migueloller Ah nice, I wasn't sure if setting that to 0 would work. I imagine the cookie would have expired before the redirect from the server action occurred, but it seems to work!

Reply:
Yeah, the `Set-Cookie` header will be in the redirect response and so the browser will set it for that response but it will immediately expire. This guarantees that it's seen only in one response's document